br/pkg/lightning/backend/local/engine.go

```diff
@@ -1158,6 +1157,16 @@ func (w *Writer) flush(ctx context.Context) error {
 	return nil
 }
 
+// EstimatedSize returns the estimated size of the SST file.
+func (w *Writer) EstimatedSize() uint64 {
+	if w.writer.Load() != nil {
+		return w.writerSize.Load()
```

Comment:
the read of `writer` and `writerSize` is not atomic. Can we check writerSize.Load() != 0 instead?